### ISSUE TITLE
Listener not listening not supported after Windows Update KB4019472

### ISSUE THREAD

**Issue by lmundy1163** [stascorp/rdpwrap#245]
I was successfully running on Windows 10 Home using RDP Wrapper version 1.6.1.  Yesterday (5/9/2017) windows update "2017-05 Cumulative Update for Windows 10 Version 1607 for x86-based Systems (KB4019472)" installed, and now the listener is reporting "not listening, not supported" on the RDP Wrapper Configuration.

Have run update.bat and it reports everything up to date.

Windows  version 1607 (OS Build 14393.1198)

termsrv.dll info:
![image](https://cloud.githubusercontent.com/assets/28601421/25906424/d3eea5e0-3561-11e7-8f36-12397ed2869e.png)

**Comment by ViperTech2016** [comment]
Same issue here.  Fresh new Windows 10 Home install as of 05-10-2017.

Not listening, fully supported
![rdpconf screenshot 20170510](https://cloud.githubusercontent.com/assets/28612993/25927776/e64ba7c2-35bd-11e7-922a-58fbd53f3845.png)

Build Info is 15063.296 Version 1703 Windows 10 Home

termsrv.dll file
[termsrv.zip](https://github.com/stascorp/rdpwrap/files/991624/termsrv.zip)

Really great program, use it a lot. Looking forward to a solution.

**Comment by binarymaster** [comment]
![image](https://cloud.githubusercontent.com/assets/578406/25936289/b76466c0-362f-11e7-94d0-eb810bff88c1.png)

I already have all those `termsrv.dll` files, no need to post them.

And there's really nothing new, everybody posting same screenshots.

Issue is already known and ticket is open!

**Comment by binarymaster** [comment]
@lmundy1163 wrote:

> Windows version 1607 (OS Build 14393.1198)

Please upload this `termsrv.dll` file.

**Comment by PocketSam** [comment]
Seems like it's what you need http://sf.it-fox.ru/f/a4b959308c/

**Comment by polreclame** [comment]
Hi,

i also have the same problem, running for a few year the RDP wrapper without any problems,
since the Update KB4019472, now it says listening / not supported. I've allready updated the ini file. it is the 2017-05-03 version.
 
![untitled-4](https://cloud.githubusercontent.com/assets/28623230/25944889/2cc954c4-3645-11e7-9dcd-411e1e955076.jpg)

**Comment by NanoByte011** [comment]
Same issue for me as well, have not been able to find a solution yet.

**Comment by bidinhos** [comment]
Same problem here with Windows 10 Pro x86 Version 1607 Build 14393.1198.
No solution yet

**Comment by ethanjansen** [comment]
I have also been having issues after updating to version 1703 on Windows 10 Home x64. However, RDP Wrapper still works on Windows 10 Pro.

**Comment by lmundy1163** [comment]
Copy of termsrv as requested

**Comment by lmundy1163** [comment]
Not sure if this zip attached on not.

**Comment by binarymaster** [comment]
> Not sure if this zip attached on not.

It's not.

**Comment by lmundy1163** [comment]
I just receive a message saying "We don't support that file type".  The file is a zip which is on the list of supported types.  Is there somewhere else I can upload it?  Contact me privately if necessary.

**Comment by bidinhos** [comment]
Here is a copy of termsrv
[termsrv.dll.zip](https://github.com/stascorp/rdpwrap/files/994736/termsrv.dll.zip)

**Comment by PocketSam** [comment]
I guess we should support RDPwrapper project, but I didn't find a way how to do that.

**Comment by PocketSam** [comment]
lmundy1163, seemes like zipped dll files are OK.

**Comment by binarymaster** [comment]
@bidinhos I already have that one.

Now I need x64 DLL.

**Comment by tzac** [comment]
How can I copy this dll in system32. I have administrators rights and still can't overwrite old dll.

**Comment by binarymaster** [comment]
@tzac don't do this.

**Comment by tzac** [comment]
@binarymaster Why ? Not working any more ? Or ........

**Comment by binarymaster** [comment]
What is your Windows processor architecture? 32-bit or 64-bit?

**Comment by tzac** [comment]
32-bit

**Comment by binarymaster** [comment]
Then you should wait until I resolve this issue. For now I require 64-bit `termsrv.dll` file for research.

**Comment by binarymaster** [comment]
@lmundy1163 try renaming .zip file to .txt, sometimes it helps.

**Comment by tzac** [comment]
I also have 64-bit termsrv.dll too. I can share it on onedrive cloud if you need it

**Comment by NanoByte011** [comment]
binarymaster, the zip in my post is 64-bit, but it's a newer version than all posting in here.

**Comment by binarymaster** [comment]
@NanoByte011 I know, that's the reason why I deleted your post.

Your issue is #194, not this one.

**Comment by NanoByte011** [comment]
ah right, sorry @binarymaster

**Comment by binarymaster** [comment]
@tzac do it please :)

**Comment by PocketSam** [comment]
Windows 10 x64 Corporate RUS
[termsrv.zip](https://github.com/stascorp/rdpwrap/files/995439/termsrv.zip)

**Comment by binarymaster** [comment]
@PocketSam this is `10.0.14393.0`, but I need  `10.0.14393.1198`.

**Comment by tzac** [comment]
@binarymaster Here it is 
[termsrv.zip](https://github.com/stascorp/rdpwrap/files/995486/termsrv.zip)

**Comment by binarymaster** [comment]
This one is 32-bit too, not 64.

**Comment by tzac** [comment]
Uh yes, my name is Stevie, Stevie Wonder, sorry

**Comment by freichar** [comment]
just checked the termsvr versions in 3 of my Win10 Pro-64bit boxes and all of them are having the version 10.0.14393.0 not a single one 10.0.14393.1198
![grafik](https://cloud.githubusercontent.com/assets/28619444/25990774/3c3a85ce-3701-11e7-809e-95825a71412e.png)

**Comment by binarymaster** [comment]
Then this issue is not your issue ;)

**Comment by freichar** [comment]
ahhh, you are right. I just realised I only have the problem on aa 32-bit box.

**Comment by astro075** [comment]
Hi 

Thanks for support. 

I have two computer 32with the same problem.
I tried to play with the temsrv.dll file (But in the wrong folder (WinSxS).
And now, on one computer I have:

Service state: Running  Ver.  **N/A**
Listener state: Not listening
I think it could be be a problem later, or not?
I try to fix it with SFC /SCANNOW

**Comment by pagslol** [comment]
Same issue, screenshot of files attached
![settings](https://cloud.githubusercontent.com/assets/28595552/25998266/957d7b3e-3763-11e7-8ca2-266e5a7d18a2.jpg)
![termsrv](https://cloud.githubusercontent.com/assets/28595552/25998270/98324b5c-3763-11e7-990c-92d6cd3fde43.jpg)
![rdp1 6 1](https://cloud.githubusercontent.com/assets/28595552/25998271/9a321a90-3763-11e7-9669-7a5996328d3b.jpg)

**Comment by rfiori** [comment]
Hi,
Here is a termsrv

.dll 64-bit  10.0.14393.1198


![image](https://cloud.githubusercontent.com/assets/4339513/25998400/2a4195ec-36f7-11e7-91b9-abda64b4d000.png)

**Comment by binarymaster** [comment]
827 KB is the size of 32-bit DLL.

**Comment by rfiori** [comment]
Hi @binarymaster ,

In 64-bit OS the **termsrv.dll** is other version see below

**Right Click** in "termsrv zip_" image below and **"Save Link As"** **(.zip)** in you PC.
![termsrv zip_](https://cloud.githubusercontent.com/assets/4339513/25999859/17d89864-36fd-11e7-86be-ceb00c596000.JPG)


![image](https://cloud.githubusercontent.com/assets/4339513/25999701/88ffda6c-36fc-11e7-8714-e7b6960b7baa.png)

**Comment by binarymaster** [comment]
@rfiori this is not `10.0.14393.1198`.

**Comment by rfiori** [comment]
@binarymaster - on 64-bit Windows version Microsoft send other version. I think only in 32-bit the version is `10.0.14393.1198`. This's not good.


![capturar](https://cloud.githubusercontent.com/assets/4339513/26000132/1e50f49c-36fe-11e7-903b-0ca5cf7fa127.JPG)

**Comment by binarymaster** [comment]
Hm... very interesting.

Since `10.0.14393.0` is already added to INI file, RDP Wrapper should work with it.

**Comment by rfiori** [comment]
Hi, @astro075  
Uninstall the update `KB4019472` in Windows update. It;s work to me.

**Comment by tzac** [comment]
This tread is open for 2017-05 Cumulative Update for Windows 10 Version 1607 for **x86-based Systems** (KB4019472) 

**32 - bit windows version**

**Comment by rfiori** [comment]
> Hm... very interesting.
> Since 10.0.14393.0 is already added to INI file, RDP Wrapper should work with it.

@binarymaster 
By analyzing, the versions are equal in 64/32 bit. But after the update KB4019472 on 32-bit version the termsrv.dll is changed to `10.0.14393.119`

![capturar](https://cloud.githubusercontent.com/assets/4339513/26001183/e716518a-3701-11e7-8daa-2aa3f3ef73ab.JPG)

**Comment by tzac** [comment]
@rfiori said: "Uninstall the update KB4019472 in Windows update. It worked for me - Until resolve this problem."

@rfiori Tnk's. Worked for me too.

**Comment by domilium** [comment]
i have no KB4019472 - was the update different for Canada?

**Comment by astro075** [comment]
@Ricardo FM

Thank you very much!

Works again. But still have

Service State: Running. Ver. N/A

😉 But works.

**Comment by esmoilo** [comment]
1. Use System Restore Explorer and mount from restore point. Copy to desktop the file termsvr.dll in windows/system32/
2. Change owner and provide User permissions in file %systemroot%/system32/termsvr.dll (this file is new)
3. Rename %systemroot%/system32/termsvr.dll to %systemroot%/system32/termsvr.dll.bak
4. Copy or move termsvr.dll in desktop to %systemroot%/system32 folder.
5. Restart your computer.

Not ideal, but until a solution appears, it works.

**Comment by JoseMaGC** [comment]
1. Use System Restore Explorer and mount from restore point. Copy to desktop the file termsvr.dll in windows/system32/
2. Change owner and provide User permissions in file %systemroot%/system32/termsvr.dll (this file is new)
3. Rename %systemroot%/system32/termsvr.dll to %systemroot%/system32/termsvr.dll.bak
4. Copy or move termsvr.dll in desktop to %systemroot%/system32 folder.
5. Restart your computer.

Not ideal, but until a solution appears, it works.

**Comment by EugeneDae** [comment]
Had to uninstall KB4019473 for rdpwrap to work as intended.

**Comment by binarymaster** [comment]
You don't need to uninstall this update, support for this termsrv version has been added in https://github.com/stascorp/rdpwrap/commit/890006de99625c194aa48dfd27c5c9a53bdb656e.

**Comment by astro075** [comment]
Yes! I confirm that. And @binarymaster: Many thanks, you rocks!

**Comment by EugeneDae** [comment]
@binarymaster This is great! Would you mind publishing the binary somewhere, possibly marked a "nightly" or "beta" build?

**Comment by binarymaster** [comment]
This is not needed since RDP Wrapper have update function.

**Comment by EugeneDae** [comment]
@binarymaster I can now confirm that RDPWrap works perfectly fine with KB4019473. My initial comment was incorrect. Bravo, Stas! Much appreciated!

**Comment by veggia** [comment]
I've the same issue with Windows 10 x64 ver 15063. Please help me

**Comment by roughwood** [comment]
I am still having issues, running update doesn't solve the problem.

![capture](https://cloud.githubusercontent.com/assets/7537179/26131107/fbb09dc8-3a65-11e7-9eb2-5e9287fc67fa.PNG)

**Comment by binarymaster** [comment]
Because this issue is not related to your issue, please don't post anything about version 15063 here.

Creators update issue is #194.

**Comment by kibeb** [comment]
@binarymaster 
> closed this in 890006d 13 days ago

Does this update fix problem with Windows 10 Home 1607 (OS Build 14393.1198) x64 with KB4019472 [1] installed too? 
Because after updating mine still says "Listener state: Not listening". :(
TIA.

(If EugeneDae and possibly others have KB4019473 installed, they are on OS Build 10586.916, not 14393.1198 as lmundy1163 who opened this issue, IMHO.)

_Ref. 1.: 2017-05 Cumulative Update for Windows 10 Version 1607 for x64-based Systems (KB4019472)_

**Comment by IMissXP** [comment]
Here is the solution that worked for me:
1) Completely uninstall any versions of RDP Wrap PRIOR to version 1.6.2 using the version's "uninstall.bat". Using "update.bat" did not work for me. I had to complete uninstall.
2) Reboot computer.
3) Install RDPWrap-v1.6.2.
Done.
I had 1.6.1 running on Windows 10 Home 32bit. Everything worked fine. Then I installed the Windows 10 Creator's Update, and I started having the "Listener state: Not listening" problem. I tried installing RDP Wrap v1.6.2 over 1.6.1, thinking the old version would be overwritten, but still no fix to the problem. Following the steps above fixed it right away.
